### PR TITLE
[native] Add 'aggregation_spill_memory_threshold' session property.

### PR DESCRIPTION
**Pull request by spershin**
Test plan - TBD.

Add 'aggregation_spill_memory_threshold' session property.
Replace mentions of 'Prestissimo' by 'Presto Native Execution'.
Move the non-native session property declarations out of TODO comment.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by highker — APPROVED** [review]

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java [hunk 1]
```diff
@@ -1356,6 +1352,16 @@ public SystemSessionProperties(
...
                         featuresConfig.isOptimizeMultipleApproxPercentileOnSameFieldEnabled(),
                         false),
+                booleanProperty(
+                        NATIVE_SIMPLIFIED_EXPRESSION_EVALUATION_ENABLED,
+                        "Native Execution only. Enable simplified path in expression evaluation",
+                        false,
+                        false),
+                integerProperty(
+                        NATIVE_AGGREGATION_SPILL_MEMORY_THRESHOLD,
+                        "Native Execution only. The max memory that a final aggregation can use before spilling. If it 0, then there is no limit",
```
Comment (mbasmanova) [inline presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java:1362]: typo: "it 0" -> "0" or "it is 0"

**Review by mbasmanova — APPROVED** [review]
@spershin Thanks.

File: presto-native-execution/presto_cpp/docs/develop.rst [hunk 2]
```diff
@@ -2,7 +2,7 @@
 Developer Guide
 ***************
 
-This guide is intended for Prestissimo power users and developers.
+This guide is intended for Presto Native Execution power users and developers.
```
[the remark this answers is not in the document]
Comment (spershin) [reply presto-native-execution/presto_cpp/docs/develop.rst:5]: I trying to find some time for it. :)

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java (hunk 1 above)
Comment (spershin) [reply presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java:1362]: Copied it from the presto_cpp :)

**Pull request merged**